src/Microsoft.NetCore.Analyzers/UnitTests/Performance/MutableStructsShouldNotBeUsedForReadonlyFieldsTests.Fixer.cs

```diff
@@ -0,0 +1,211 @@
+    {
+        [Fact]
+        public async Task CSharpReadonlyFieldsOfKnownMutableTypes_RemovesReadonlyModifier()
+        {
+            var initial = @"
+using System;
+using System.Threading;
+using System.Runtime.InteropServices;
+
+public class C
+{
+    private [|readonly|] SpinLock _sl = new SpinLock();
+    private [|readonly|] GCHandle _gch = new GCHandle();
+
+    private [|readonly|] SpinLock _sl_noinit;
+    private [|readonly|] GCHandle _gch_noinit;
+}
+";
+
+            var expected = @"
+using System;
+using System.Threading;
+using System.Runtime.InteropServices;
+
+public class C
+{
+    private SpinLock _sl = new SpinLock();
+    private GCHandle _gch = new GCHandle();
+
+    private SpinLock _sl_noinit;
+    private GCHandle _gch_noinit;
+}
+";
+            await CSharpCodeFixVerifier<CSharpMutableStructsShouldNotBeUsedForReadonlyFieldsAnalyzer, CSharpMutableStructsShouldNotBeUsedForReadonlyFieldsFixer>
+                .VerifyCodeFixAsync(initial, expected);
+        }
+
+        [Fact]
+        public async Task CSharpWritableFieldsOfKnownMutableType_NoDiagnosticNoFix()
+        {
+            var initial = @"
+using System;
+using System.Threading;
+using System.Runtime.InteropServices;
+
+public class C
+{
+    private SpinLock _sl = new SpinLock();
+    private GCHandle _gch = new GCHandle();
+
+    private SpinLock _sl_noinit;
+    private GCHandle _gch_noinit;
+}
+";
+
+            var expected = @"
+using System;
+using System.Threading;
+using System.Runtime.InteropServices;
+
+public class C
+{
+    private SpinLock _sl = new SpinLock();
+    private GCHandle _gch = new GCHandle();
+
+    private SpinLock _sl_noinit;
+    private GCHandle _gch_noinit;
+}
+";
+            await CSharpCodeFixVerifier<CSharpMutableStructsShouldNotBeUsedForReadonlyFieldsAnalyzer, CSharpMutableStructsShouldNotBeUsedForReadonlyFieldsFixer>
+                .VerifyCodeFixAsync(initial, expected);
+        }
+
+        [Fact]
+        public async Task CSharpReadonlyFieldsOfCustomType_NoDiagnosticNoFix()
+        {
+            var initial = @"
+using System;
+using System.Threading;
+using System.Runtime.InteropServices;
+
+public struct S 
+{
+}
+
+public class C
+{
+    private readonly S _s = new S();
+}
+";
+
+            var expected = @"
+using System;
+using System.Threading;
+using System.Runtime.InteropServices;
+
+public struct S 
+{
+}
+
+public class C
+{
+    private readonly S _s = new S();
+}
+";
+            await CSharpCodeFixVerifier<CSharpMutableStructsShouldNotBeUsedForReadonlyFieldsAnalyzer, CSharpMutableStructsShouldNotBeUsedForReadonlyFieldsFixer>
+                .VerifyCodeFixAsync(initial, expected);
+        }
+
+        [Fact]
+        public async Task BasicReadonlyFieldsOfKnownMutableTypes_RemovesReadonlyModifier()
+        {
+            var initial = @"
+Imports System.Threading
+Imports System.Runtime.InteropServices
+
+Public Class Class1
+    Public [|ReadOnly|] _sl As SpinLock = New SpinLock()
```

Comment:
Can you add VB unit test with field declaration that using AsNew clause. For example, `Private Readonly _sl As New SpinLock()`